File: src/main/java/xyz/aflkonstukt/purechaos/PackPriority.java
package xyz.aflkonstukt.purechaos;

import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.server.packs.PackType;
import net.minecraft.server.packs.PathPackResources;
import net.minecraft.server.packs.repository.*;
import net.neoforged.bus.api.IEventBus;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.ModList;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.fml.common.Mod;
import net.neoforged.neoforge.event.AddPackFindersEvent;

@EventBusSubscriber(bus = EventBusSubscriber.Bus.MOD, modid = PurechaosMod.MODID)
public class PackPriority {
    public PackPriority(IEventBus modEventBus) {
        modEventBus.register(this);
    }

    @SubscribeEvent
    public static void addPackFinders(AddPackFindersEvent event) {
        event.addPackFinders(ResourceLocation.fromNamespaceAndPath(PurechaosMod.MODID, "pmo"), PackType.CLIENT_RESOURCES, Component.literal("PMOs"), PackSource.BUILT_IN, true, Pack.Position.TOP);
    }
}

File: pure-spice/src/main/kotlin/xyz/aflkonstukt/spice/special/PackPriority.kt
package xyz.aflkonstukt.spice.special

import net.minecraft.network.chat.Component
import net.minecraft.resources.ResourceLocation
import net.minecraft.server.packs.PackType
import net.minecraft.server.packs.repository.Pack
import net.minecraft.server.packs.repository.PackSource
import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.fml.common.EventBusSubscriber
import net.neoforged.neoforge.event.AddPackFindersEvent
import xyz.aflkonstukt.purechaos.PurechaosMod
import xyz.aflkonstukt.spice.Spice

@EventBusSubscriber(bus = EventBusSubscriber.Bus.MOD, modid = Spice.ID)
object PackPriority {
    @SubscribeEvent
    fun addPackFinders(event: AddPackFindersEvent) {
        event.addPackFinders(
            ResourceLocation.fromNamespaceAndPath(Spice.ID, "pmo"),
            PackType.CLIENT_RESOURCES,
            Component.literal("PMOs"),
            PackSource.BUILT_IN,
            true,
            Pack.Position.TOP
        )
    }
}